In [5]:
# CARGAR LOS DATOS
import pandas as pd
import numpy as np

df = pd.read_excel("usoDiario.xlsx", header=17)
df = df.dropna(how="all")

df.columns

Index(['xi(uso diario redes sociales)', 'fi', 'Fi', 'fri', 'Fri', 'pi%', 'Pi%',
       'media aritmética', 'moda', 'mediana', 'C.asimetría', 'Curtosis', 'mc',
       'Varianza', 'Desvío estándar', 'C.variación'],
      dtype='str')

In [6]:
usoDiario = df["xi(uso diario redes sociales)"]

In [7]:
df = df.dropna(subset=["xi(uso diario redes sociales)"]).head(6) #ELIMINAMOS LAS OTRAS COLUMNAS

# Aseguramos que las frecuencias absolutas sean números enteros
fi = df["fi"].astype(int).values
total_n = fi.sum() # Debería dar 200

# Definimos los intervalos y las marcas de clase (mc) para el cálculo de datos agrupados
intervalos = df["xi(uso diario redes sociales)"].values
mc = np.array([1, 3, 5, 7, 9, 11]) # Puntos medios de los intervalos (0-2, 2-4, etc.) Como no sabemos exactamente los valores individuales, 
#tomas el punto medio del intervalo como representante de los valores individuales :)
#para (2-4) es (2+4)/2 = 3, y lo mismo sucesivamente


# ==========================================
# RECONSTRUCCIÓN DE LA TABLA EN PYTHON
# ==========================================
tabla = pd.DataFrame() #cremos un DataFrame nuevo y le vamos agregando columnas y sus valores

tabla["Intervalo"] = intervalos
tabla["fi"] = fi #Frecuencia absoluta, que nos dice cuántas veces ocurre
tabla["Fi"] = tabla["fi"].cumsum() #la función .cumsum() hace la suma acumulada (suma las frecuencias absolutas anteriores)
tabla["fri"] = tabla["fi"] / total_n #tomamos cada frecuencia absoluta (fi) y la dividimos por el total (total_n). Eso nos dice qué proporción de la muestra pertenece a esa clase (ej. 0.15)
tabla["Fri"] = tabla["fri"].cumsum() # es lo mismo que la frecuencia absoluta acumulada, pero con las relativa
tabla["pi%"] = tabla["fri"] * 100 #s la frecuencia relativa multiplicada por 100 para tenerlo en formato porcentaje (siguiendo el ej. anterior, 15%)
tabla["Pi%"] = tabla["pi%"].cumsum() #es la frecuencia relativa acumulada multiplicada por 100, o así también la suma consecutiva de pi%

print("--- TABLA DE FRECUENCIAS COMPLETA (Fase 4) ---")
print(tabla.to_string(index=False))
print("-" * 50)

# ==========================================
# CÁLCULOS ESTADÍSTICOS PARA DATOS AGRUPADOS
# ==========================================

# calculamos media Aritmética (Suma de mc * fi / N)
media = np.sum(mc * fi) / total_n #para calcular el promedio con datos agrupados, multiplicas cada marca de clase (mc) 
#por su frecuencia (fi), sumas todos esos resultados, y luego lo divides por el total de la muestra (total_n)

# calculamos varianza y Desvío Estándar para datos agrupados

varianza = np.sum(fi * (mc - media) ** 2) / total_n #mide qué tan "esparcidos" están los datos alrededor de la media. Para cada intervalo, se
#toma la marca de clase, se resta la media, se eleva el resultado al cuadrado, lo multiplicas por la frecuencia de ese intervalo, sumas todo y divides por el total.

desvio_estandar = np.sqrt(varianza)#es simplemente la raíz cuadrada (np.sqrt) de la varianza, es un poquito mas facil de entender porq syu unidad de medida es en horas.

coeficiente_variacion = (desvio_estandar / media) * 100#mide la relación entre el desvío estándar y la media como un porcentaje, 
#sirve para comparar la dispersión de dos conjuntos de datos diferentes


print("\n--- ESTADÍSTICAS GENERADAS EN PYTHON ---")
print(f"Media aritmética (medida pertinente de comparación): {media:.2f} horas diarias")
print(f"Varianza: {varianza:.2f}")
print(f"Desvío estándar: {desvio_estandar:.2f} horas")
print(f"Coeficiente de variación: {coeficiente_variacion:.2f}%")

--- TABLA DE FRECUENCIAS COMPLETA (Fase 4) ---
Intervalo  fi  Fi   fri   Fri  pi%   Pi%
    0 - 2  30  30 0.150 0.150 15.0  15.0
    2 - 4  50  80 0.250 0.400 25.0  40.0
    4 - 6  43 123 0.215 0.615 21.5  61.5
    6 - 8  34 157 0.170 0.785 17.0  78.5
   8 - 10  20 177 0.100 0.885 10.0  88.5
  10 - 12  23 200 0.115 1.000 11.5 100.0
--------------------------------------------------

--- ESTADÍSTICAS GENERADAS EN PYTHON ---
Media aritmética (medida pertinente de comparación): 5.33 horas diarias
Varianza: 9.71
Desvío estándar: 3.12 horas
Coeficiente de variación: 58.47%


## Cuarta etapa: Contrastación de datos (ANÁLISIS)
Para evaluar y contrastar el comportamiento del uso de redes sociales entre la muestra recolectada por el grupo en la primera etapa y la base de datos provista por la cátedra en la cuarta etapa, se seleccionó la Media Aritmética como la medida de tendencia central más pertinente. Asimismo, se utilizó el Coeficiente de Variación (CV) con el fin de analizar la dispersión y la homogeneidad de los datos en ambas muestras.

Cabe destacar que, para realizar una comparación equivalente, se procedió a promediar la media de la primera etapa (calculada originalmente en 31 horas semanales) dividiéndola por los siete días de la semana, obteniendo así una media comparable de 4.43 horas diarias.

A continuación, se presenta el análisis comparativo basado en las respuestas a los interrogantes planteados:

* ¿En qué grupo se pasa más tiempo en las redes sociales?
Al contrastar las medias aritméticas diarias, se observa que en la Cuarta etapa (uso diario redes sociales) se registra un mayor consumo de pantallas, alcanzando un promedio de 5.33 horas diarias, frente a las 4.43 horas diarias obtenidas en la encuesta de la Primera etapa.

* ¿Qué grupo es más "parejo" (homogéneo) en su consumo?
El grupo estadísticamente más homogéneo en sus hábitos de consumo es el de la Cuarta etapa, ya que presenta un Coeficiente de Variación significativamente menor (58.47%), en comparación con el 122.41% de la Primera etapa. Esto significa que en los datos de la cuarta etapa las respuestas de los estudiantes son más similares entre sí, mientras que en la primera etapa existe una dispersión extrema (con casos de consumo muy dispares entre los encuestados).

* ¿Ambos grupos superan las 3 horas diarias recomendadas por la OMS?
Sí, ambos grupos superan ampliamente el límite máximo saludable de 2 a 3 horas diarias sugerido por la Organización Mundial de la Salud (OMS) para adolescentes y jóvenes. Particularmente en la cuarta etapa, el promedio de 5.33 horas diarias representa casi el doble del umbral estipulado. Esto pone en evidencia una problemática generalizada de sobreexposición a las pantallas en los entornos estudiados, lo que incrementa significativamente el riesgo de desarrollar conductas asociadas a la adicción a internet y un desequilibrio en el bienestar digital